extensions/sql-migration/src/scripts/SHIR-auto-configuration.ps1

```diff
@@ -0,0 +1,609 @@
+				Write-OutputAndLog "$packageName package is already installed."
+				return $true
+			}
+			else {
+				Write-OutputAndLog "$packageName package is not installed."
+			}
+		}
+		catch {
+			Write-ErrorAndLog -exception $_.Exception
+			return $false
+		}
+
+		# add necessary arguments to install quietly with no UI
+		# for no UI /qn, basic UI /qb, reduced UI /qr, full UI /qf
+		$argsx = @('/i', $packagePath, '/quiet', '/qb')
+
+		# execute installation process
+		Write-OutputAndLog "Starting $packageName package installation..."
+		$process = Start-Process msiexec -Wait -NoNewWindow -ErrorAction SilentlyContinue -ArgumentList $argsx
+		Write-OutputAndLog ("MSI Installation process exited with code: " + $process.ExitCode)
+		# after installation update the environment variable for the powershell session
+		$env:Path = [System.Environment]::GetEnvironmentVariable("Path", [System.EnvironmentVariableTarget]::Machine)
+
+		# Verifying the installation
+		Write-OutputAndLog "Verifying $packageName package installation..."
+		try {
+			$installationResult = & $installationVerificationCallback -ErrorAction SilentlyContinue
+			if ($null -ne $installationResult -and $installationResult -eq $true) {
+				Write-OutputAndLog "$packageName package installed sucessfully."
+				return $true
+			}
+			else {
+				Write-OutputAndLog "Failed to verify $packageName installation."
+				return $false
+			}
+		}
+		catch {
+			Write-ErrorAndLog -exception $_.Exception
+			# Only throw the error incase there is a valid verification command
+			if (-not [string]::IsNullOrEmpty($installationVerificationCallback.ToString())) {
+				Write-ErrorAndLog "Installation Error: Installation of package $packageName failed verification"
+			}
+			return $false
+		}
+	}
+	catch {
+		Write-ErrorAndLog -exception $_.Exception
+		return $false
+	}
+	return $true
+}
+
+#################################### Register Node to DMS ###################################
+Function Register-IntegrationRuntime {
+	<#
+    .DESCRIPTION
+    Configures the SHIR with the auth key. It registers the IR to the DMS using the authentication keys.
+
+    .OUTPUTS
+    None.
+    #>
+	param (
+		[Parameter()]
+		[string]$AuthKey1,
+
+		[Parameter()]
+		[string]$AuthKey2
+	)
+
+	if ([string]::IsNullOrWhitespace($AuthKey1) -and [string]::IsNullOrWhitespace($AuthKey2)) {
+		Write-ErrorAndLog "You need to provide atleast one authentication key to configure SHIR."
+		return
+	}
+
+	$azDataMigrationModule = Get-Module -ListAvailable -Name Az.DataMigration
+
+	if (-not $azDataMigrationModule) {
+		# Install the NugetProvider for Az.DataMigration
+		$nuget = Get-PackageProvider -Name NuGet -ErrorAction SilentlyContinue
+
+		if ($null -eq $nuget -or $nuget.Version -lt $Global:MinRequiredNuGetVersion) {
+			Install-PackageProvider -Name NuGet -MinimumVersion $Global:MinRequiredNuGetVersion -Force
+		}
+
+		# Install the Az.DataMigration module
+		Install-Module -Name Az.DataMigration -Force
```

Comment:
Should this be under try block so the exception could be caught in case of failure?

Reply:
Done